benches/README.md

```diff
@@ -0,0 +1,54 @@
+# Miden VM Hash Functions 
+In Miden VM, we make use of different hash functions. Some of these are "traditional" hash functions, like `Blake3`, which are optimized for out of STARK performance, while others are algebraic hash functions, like `Rescue Prime`, and are more optimized for a better performance inside the STARK. In what follows, we benchmark several such hash functions and compare against other constructions that are used by other proving systems. More precisely, we benchmark:
+
+## Rescue Prime
+As specified [here](https://eprint.iacr.org/2020/1143) and implemented [in](https://github.com/novifinancial/winterfell/blob/46dce1adf0/crypto/src/hash/rescue/rp64_256/mod.rs).
+
+## Rescue Prime Optimized
+As specified [here](https://eprint.iacr.org/2022/1577) and implemented in this crate.
+
+## BLAKE3
+As specified [here](https://github.com/BLAKE3-team/BLAKE3-specs/blob/master/blake3.pdf) and implemented in this crate.
+
+## SHA256
+As specified [here](https://csrc.nist.gov/csrc/media/publications/fips/180/4/final/documents/fips180-4-draft-aug2014.pdf) and implemented [in](https://github.com/novifinancial/winterfell/blob/46dce1adf0/crypto/src/hash/sha/mod.rs).
+
+## Poseidon
+As specified [here](https://eprint.iacr.org/2019/458.pdf) and implemented (without vectorized instructions) [in](https://github.com/mir-protocol/plonky2/blob/main/plonky2/src/hash/poseidon_goldilocks.rs).
+
+# Comparison and Instructions
+
+## Comparison
+We benchmark the above hash functions using two scenarios. The first is a 2-to-1 $(a,b)\mapsto h(a,b)$ hashing where both $a$, $b$ and $h(a,b)$ are $4$ field elements i.e. 256-bit.
```

Comment:
We probably need to re-phrase this as not all hash functions work with field elements.

Reply:
Done, let me know if it is still not clear.